x-pack/legacy/plugins/siem/public/containers/case/configure/use_connectors.tsx

```diff
@@ -0,0 +1,109 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import { useState, useEffect, useCallback } from 'react';
+
+import { useStateToaster, errorToToaster } from '../../../components/toasters';
+import * as i18n from '../translations';
+import { fetchConnectors, patchConfigConnector } from './api';
+import { CasesConfigurationMapping, Connector } from './types';
+
+export interface ReturnConnectors {
+  loading: boolean;
+  connectors: Connector[];
+  refetchConnectors: () => void;
+  updateConnector: (connectorId: string, mappings: CasesConfigurationMapping[]) => Promise<unknown>;
+}
+
+export const useConnectors = (): ReturnConnectors => {
+  const [, dispatchToaster] = useStateToaster();
+  const [loading, setLoading] = useState(true);
+  const [connectors, setConnectors] = useState<Connector[]>([]);
+
+  const refetchConnectors = useCallback(async () => {
```

Comment:
what do you think about adopting the closure async/await pattern we discussed in my PR? https://dev.to/n1ru4l/homebrew-react-hooks-useasynceffect-or-how-to-handle-async-operations-with-useeffect-1fa8.

Reply:
https://github.com/elastic/kibana/pull/59358#discussion_r391170445